observer.go

```diff
@@ -69,3 +69,26 @@ func (o *compositeObserver) OnStartSpan(sp opentracing.Span, operationName strin
 	}
 	return &compositeSpanObserver{observers: spanObservers}
 }
+
+// compositeSpanObserver is a dispatcher to other span observers
+type compositeSpanObserver struct {
```

Comment:
nit: I prefer defining the struct at the top of the file before it gets used in the other functions.